src/coreclr/tools/aot/ILCompiler.RyuJit/JitInterface/CorInfoImpl.Llvm.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using ILCompiler;
+using Internal.TypeSystem;
+
+namespace Internal.JitInterface
+{
+    public unsafe sealed partial class CorInfoImpl
+    {
+        [ThreadStatic]
+        private static CorInfoImpl _thisStatic;
+
+        [UnmanagedCallersOnly]
+        public static byte* getMangledMethodName(IntPtr thisHandle, CORINFO_METHOD_STRUCT_* ftn)
+        {
+            //var _this = GetThis(thisHandle); // TODO: this doesn't work, but how does it cope anyway with this being moved by the GC?
```

Comment:
> but how does it cope anyway with this being moved by the GC?

The IntPtr points to stack location where `CorInfoImpl` reference is stored. 

JIT/EE interface assumes that the JIT does not store the this pointer in a static.  `registerLlvmCallbacks` stores the this pointer in a static. I suspect that it is why it does not work.

Reply:
Thanks for the explanation, have done something similar so this now works without the static.